### PR TITLE
Custom settings docs

### PR DESCRIPTION
**Pull request by Thomas-James-Rose**
WIP for documentation around custom settings and settings callbacks.

### CODE REVIEW

File: docs/gameplay-components/custom-settings.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+In `src/main.js` subscribe to the settings channel by calling `signal.bus.subscribe(subscription)` where `subscription` is an object with the following properties:
+- `channel`: This should be set to settingsChannel which is imported at the top of `src/main.js`.
+- `name`: This should correspond to the key of the setting that the callback is being set for.
+- `callback`: A callback function that will execute whenever the setting is changed from the settings dialog box.
+
+Example usage:
+
+```javascript
+signal.bus.subscribe({
+    channel: settingsChannel,
```
Comment (Thomas-James-Rose) [inline docs/gameplay-components/custom-settings.md]: I think I've already made brief mention of this on Friday to some people but if settings should always use the settings channel we should probably do a small bit of refactoring at some point so we don't need agencies to add this. Maybe we should just have some function they can call like `addSettingsCallback("settingName", () => { do stuff });`

Comment (mcmilj) [reply docs/gameplay-components/custom-settings.md]: Funny you should say this, Sean and I were just talking about it!

Comment (Thomas-James-Rose) [reply docs/gameplay-components/custom-settings.md]: Great minds and all that. 😆

Comment (SeanHolden) [reply docs/gameplay-components/custom-settings.md]: Sounds good to me. Would be best being another ticket 👍

Comment (Thomas-James-Rose) [reply docs/gameplay-components/custom-settings.md]: Yeah, I agree. We'll have to remember to update these docs as a part of it though.

**Pull request merged**